### PR TITLE
feat: add cloud logs policies

### PR DESCRIPTION
**Pull request by maheshwarishikha**
### Description

Add cloud logs policies

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Add cloud logs policies

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by maheshwarishikha** [comment]
This time PR failed in EN integration code - 
```
2024/10/25 09:00:55 Terraform apply | module.observability_instances.module.cloud_logs[0].ibm_logs_outgoing_webhook.en_integration["1"]: Creation complete after 4s [id=eu-de/e765aee9-9bd6-45f5-9493-cf0b875ffb80/968730df-7b2e-45fa-97bd-4155f2f61789]
 2024/10/25 09:00:55 Terraform apply | 
 2024/10/25 09:00:55 Terraform apply | Error: ---
 2024/10/25 09:00:55 Terraform apply | id: terraform-693866e6
 2024/10/25 09:00:55 Terraform apply | summary: 'CreateOutgoingWebhookWithContext failed: No authentication information in
 2024/10/25 09:00:55 Terraform apply |   RequestContext'
 2024/10/25 09:00:55 Terraform apply | severity: error
 2024/10/25 09:00:55 Terraform apply | resource: ibm_logs_outgoing_webhook
 2024/10/25 09:00:55 Terraform apply | operation: create
 2024/10/25 09:00:55 Terraform apply | component:
 2024/10/25 09:00:55 Terraform apply |   name: github.com/IBM-Cloud/terraform-provider-ibm
 2024/10/25 09:00:55 Terraform apply |   version: 1.70.1
 2024/10/25 09:00:55 Terraform apply | ---
 2024/10/25 09:00:55 Terraform apply | 
 2024/10/25 09:00:55 Terraform apply | 
 2024/10/25 09:00:55 Terraform apply |   with module.observability_instances.module.cloud_logs[0].ibm_logs_outgoing_webhook.en_integration["0"],
```

**Review by ocofaigh — CHANGES_REQUESTED** [review]
I suggest to support creating multiple policies by using 1 input variable that is a list of objects

**Comment by maheshwarishikha** [comment]
> I suggest to support creating multiple policies by using 1 input variable that is a list of objects

sure, will update the code accordingly.

File: variables.tf [hunk 1]
```diff
@@ -275,4 +275,31 @@ variable "logs_routing_tenant_regions" {
   nullable    = false
 }
 
+########################################################################
+# Cloud Logs Policies - TCO Optimizer
+#########################################################################
+
+variable "logs_policies_config" {
```
Comment (ocofaigh) [inline variables.tf:282]: suggest to call this `cloud_logs_policies`. All the other cloud logs variables start with `cloud_logs_` so lets be consistent (note that logs_routing is a different service)

File: modules/cloud_logs/variables.tf [hunk 2]
```diff
@@ -150,3 +150,124 @@ variable "skip_logs_routing_auth_policy" {
...
+# logs_policy_name -The name of the IBM Cloud Logs policy to create.
+# logs_policy_description - Description of the IBM Cloud Logs policy to create.
+# logs_policy_priority - Select priority to determine the pipeline for the logs. High (priority value) sent to 'Priority insights' (TCO pipleine), Medium to 'Analyze and alert', Low to 'Store and search', Blocked are not sent to any pipeline.
+# application_rule - Define rules for matching applications to include in the policy configuration.
+# subsystem_rule - Define subsystem rules for matching applications to include in the policy configuration.
+# log_rules - Define the log severities to include in the policy configuration.
+# archive_retention - Define archive retention.
+##############################################################################################################
+
+variable "logs_policies_config" {
```
Comment (ocofaigh) [inline modules/cloud_logs/variables.tf:166]: maybe just call this `policies`

File: modules/cloud_logs/README.md [hunk 3]
```diff
@@ -46,6 +46,20 @@ module "cloud_logs" {
       bucket_endpoint = "s3.direct.us-south.cloud-object-storage.appdomain.cloud"
     }
   }
+  # Create log policy
```
Comment (ocofaigh) [inline modules/cloud_logs/README.md:49]: this usage needs to be update

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Some minor comments. I suggest that you add the policy to the advanced example instead of basic. Thats also the example we run in PR tests

File: modules/cloud_logs/variables.tf (hunk 2 above)
Comment (maheshwarishikha) [reply modules/cloud_logs/variables.tf]: done

File: modules/cloud_logs/README.md (hunk 3 above)
Comment (maheshwarishikha) [reply modules/cloud_logs/README.md]: updated

**Comment by maheshwarishikha** [comment]
PR failed with EN related issue - 
```
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: ╷
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ Error: ---
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ id: terraform-693866e6
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ summary: 'CreateOutgoingWebhookWithContext failed: No authentication information in
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │   RequestContext'
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ severity: error
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ resource: ibm_logs_outgoing_webhook
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ operation: create
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ component:
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │   name: github.com/IBM-Cloud/terraform-provider-ibm
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │   version: 1.70.1
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ ---
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ 
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ 
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │   with module.observability_instances.module.cloud_logs[0].ibm_logs_outgoing_webhook.en_integration["1"],
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │   on ../../modules/cloud_logs/main.tf line 109, in resource "ibm_logs_outgoing_webhook" "en_integration":
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │  109: resource "ibm_logs_outgoing_webhook" "en_integration" {
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: │ 
TestRunUpgradeExample 2024-10-29T10:28:06Z logger.go:66: ╵
TestRunUpgradeExample 2024-10-29T10:28:06Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; 
```

re-running...

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**